### PR TITLE
Use AsciiDoc AST to create open v3 support

### PR DESCRIPTION
**Pull request by austek**
OpenApi v3 schema document support implemented as follows;
- OpenAPI Object (implemented)
- Info Object (implemented)
- Contact Object (implemented)
- License Object (implemented)
- Server Object (implemented)
- Server Variable Object (implemented)
- Components Object (partially implemented, needs: securitySchemes, callbacks, requestBodies, examples)
- Paths Object  (implemented)
- Path Item Object (implemented)
- Operation Object (partially implemented, needs: securitySchemes, callbacks, requestBodies, examples)
- External Documentation Object (implemented)
- Parameter Object (implemented)
- Request Body Object
- Media Type Object
- Encoding Object
- Responses Object (implemented)
- Response Object (implemented)
- Callback Object
- Example Object
- Link Object (implemented)
- Header Object (implemented)
- Tag Object (implemented)
- Reference Object (implemented)
- Schema Object (implemented)
- Discriminator Object (not-implemented)
- XML Object (not-implemented)
- Security Scheme Object (not-implemented)
- OAuth Flows Object (not-implemented)
- OAuth Flow Object (not-implemented)
- Security Requirement Object (not-implemented)

### CODE REVIEW

File: swagger2markup-swagger-v3/src/main/java/io/github/swagger2markup/OpenApiPathsSection.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        if(null == apiPaths || apiPaths.isEmpty()) return;
+
+        SectionImpl allPathsSection = new SectionImpl(document);
+        allPathsSection.setTitle(SECTION_TITLE_PATHS);
+
+        apiPaths.forEach((name, pathItem) -> {
+            pathItem.readOperationsMap().forEach(((httpMethod, operation) -> {
+                SectionImpl operationSection = new SectionImpl(allPathsSection);
+                String summary = Optional.ofNullable(operation.getSummary()).orElse("");
+                operationSection.setTitle((italicUnconstrained(httpMethod.name().toUpperCase()) + " " + monospaced(name) + " " + summary).trim());
```
Comment (RobWin) [inline swagger2markup-swagger-v3/src/main/java/io/github/swagger2markup/OpenApiPathsSection.java]: Can we change this to how it looks today?
image

File: swagger2markup-swagger-v3/src/main/java/io/github/swagger2markup/OpenApiHelpers.java [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static io.github.swagger2markup.adoc.converter.internal.Delimiters.LINE_SEPARATOR;
+
+public class OpenApiHelpers {
+
+    public static final String LABEL_DEFAULT = "Default";
```
[the remark this answers is not in the document]
Comment (austek) [reply swagger2markup-swagger-v3/src/main/java/io/github/swagger2markup/OpenApiHelpers.java]: Internationalization will be supported, that's the reason, all strings are collected in one place to make it easier to do that. Just wanted to do an initial pass for v3 support

File: swagger2markup-swagger-v3/src/main/java/io/github/swagger2markup/OpenApiPathsSection.java (hunk 1 above)
Comment (austek) [reply swagger2markup-swagger-v3/src/main/java/io/github/swagger2markup/OpenApiPathsSection.java]: Certainly can be done, I opted for the current implementation as summary is an optional value can be missing. what could be used for such scenarios?

**Comment by ewoks** [comment]
Not 100% sure about the plans, but this ticket is probably ready to be closed. Correct?

**Comment by admoca60** [comment]
Hi @RobWin !!

I could see that this feature is merged to master, but none new version of this artifact has been released from two years ago. 

Do you have any plan about doing a new release with these changes? Will be this feature available to be used with maven plugin? 

Thank you!

**Pull request merged**